apps/server/src/modules/room/api/saga/copy-room.step.ts

```diff
@@ -41,7 +41,9 @@ export class CopyRoomStep extends SagaStep<'copyRoom'> {
 		const user = await this.authorizationService.getUserWithPermissions(userId);
 		const originalRoom = await this.roomService.getSingleRoom(roomId);
 
-		const copyName = newName ?? (await this.deriveCopyName(userId, originalRoom.name));
+		const name = newName ?? originalRoom.name;
+		const copyName = await this.deriveCopyName(userId, name);
+
```

Comment:
Good point. So the new name is also derived from the new name. Could be a name that already has been given.